drawee/src/main/java/com/facebook/drawee/generic/GenericDraweeHierarchy.java

```diff
@@ -173,9 +174,13 @@ public void draw(Canvas canvas) {
         mRoundingParams,
         mResources,
         placeholderImageBranch);
+    mPlaceHolderScaleType = builder.getPlaceholderImageScaleType();
+    if (mPlaceHolderScaleType == null) {
+        mPlaceHolderScaleType = GenericDraweeHierarchyBuilder.DEFAULT_SCALE_TYPE;
+    }
```

Comment:
If we decide to do this, it would be better to change [GenericDraweeHierarchyBuilder] (https://github.com/facebook/fresco/blob/920539d9443bc3979fbd838ecd64d3356b735ee0/drawee/src/main/java/com/facebook/drawee/generic/GenericDraweeHierarchyBuilder.java#L83) than to do it here.

However, I don't think we should do this at all. This change will cause a ScaleTypeDrawable to be always created for each placeholder even though in some cases this is not necessary. E.g. color drawables (including a default placeholder which is just a transparent drawable) do not need any scaling because scaling does not affect them. This will impose an unnecessary memory overhead.

If you are specifying the placeholder programmatically after the hierarchy is being built, and you don't have the placeholder available at the time the hierarchy is being built, then you can achieve the same behavior by doing `GenericDraweeHierarchyBuilder.setPlaceholderImage(null, someScaleType)`. This will ensure that a ScaleTypeDrawable for the placeholder image gets created.